Katana/Store.swift

```diff
@@ -512,7 +546,7 @@ fileprivate extension Store {
     self.state = typedNewState
     
     // listener are always invoked in the main queue
```

Comment:
at this point, same as above here i'd say..

Reply:
eh, not really. listeners in the apps must be called from the main queue (given how we are doing things rn) because they are mainly tempura stuff. The fact that we are using a protocol to encapsulate this logic is a technical detail (that among other things make stuff easier to test) but semantically that IS the main queue